flytekit/configuration/file.py

```diff
@@ -80,10 +86,11 @@ def read_from_file(
             return None
         try:
             v = cfg.get(self)
-            if v:
+            if _exists(v):
```

Comment:
but `isinstance(False, bool)` is true, so if `v` was `False` now this will be true.  Or is `v` never a bool?

Reply:
Yep, it wasn't picking up `insecure: false` in the yaml config because `if v` would evaluate to false, so the config would just always use the default value.

Reply:
I guess the `transform` here doesn't actually do what it's intended to do before yaml will automatically parse values to the expected types